nixos/modules/tasks/filesystems.nix

```
@@ -24,13 +24,15 @@ let
 
   specialFSTypes = [ "proc" "sysfs" "tmpfs" "ramfs" "devtmpfs" "devpts" ];
 
+  nonEmptyWithoutTrailingSlash = addCheckDesc "non-empty without trailing slash" types.str
+    (s: isNonEmpty s && (builtins.match ".+/" s) == null);
```

Is there a reason for these not to start with a `/`?
```suggestion
  nonEmptyPathWithoutTrailingSlash = addCheckDesc "non-empty without trailing slash" types.path
    (s: isNonEmpty s && (builtins.match ".+/" s) == null);
```
Note that this doesn't make much sense:
* any `path` is non-empty
* the path `/` has a trailing slash but is still allowed (even before this change)

---

I know you did. I guess we can leave improving this to a different PR.